### ISSUE TITLE
nfa: support converting into a DFA and inverting 

### ISSUE THREAD

**Issue by georgiy-belyanin** [MacadamiaSolver/MacadamiaSolver#10]
After implementing #1 and adding the NFAs we are allowed to eliminate some of the existence quantifiers quite easily. Though we still don't have enough to eliminate negations and any quantifiers. The method to turn NFAs into some DFAs and allow safely inverting the final states is needed for it.

Two new functions are suggested to be implemented:
```ocaml
val make_determenistic: Nfa -> Dfa
(* This one makes DFA and inverts it. *)
val invert: Nfa -> Dfa
```

Probably, the first one doesn't need to be public. Though it's not clean yet.

**Comment by georgiy-belyanin** [comment]
Инокентий, please, check this issue and ask if you need any help

**Comment by georgiy-belyanin** [comment]
Solved in MacadamiaSolver@fda2523
